ts/features/bonus/cgn/store/reducers/otp.ts

```diff
@@ -0,0 +1,50 @@
+import { getType } from "typesafe-actions";
+import { Action } from "../../../../../store/actions/types";
+import { GlobalState } from "../../../../../store/reducers/types";
+import { NetworkError } from "../../../../../utils/errors";
+import { Otp } from "../../../../../../definitions/cgn/Otp";
+import { cngGenerateOtp } from "../actions/otp";
+import {
+  remoteError,
+  remoteLoading,
+  remoteReady,
+  remoteUndefined,
+  RemoteValue
+} from "../../../bpd/model/RemoteValue";
+
+export type CgnOtpState = {
+  data: RemoteValue<Otp, NetworkError>;
+};
+
+const INITIAL_STATE: CgnOtpState = {
+  data: remoteUndefined
+};
+
+const reducer = (
+  state: CgnOtpState = INITIAL_STATE,
+  action: Action
+): CgnOtpState => {
+  switch (action.type) {
+    case getType(cngGenerateOtp.request):
+      return {
+        ...state,
+        data: remoteLoading
+      };
+    case getType(cngGenerateOtp.success):
+      return {
+        ...state,
+        data: remoteReady(action.payload)
+      };
+    case getType(cngGenerateOtp.failure):
```

Comment:
```suggestion
    case getType(cgnGenerateOtp.failure):
```

Reply:
thanks!
changed in https://github.com/pagopa/io-app/pull/2882/commits/0c7f2ce791d125d60b89fc83901c40150c741946